web/src/pages/Upload.jsx

```diff
@@ -0,0 +1,152 @@
+import { UploadOutlined, CloudUploadOutlined } from "@ant-design/icons";
+import { Button, message, Upload, Card, Typography, Divider } from "antd";
+import { useState } from "react";
+
+const { Title, Text } = Typography;
+
+export const FileUploading = () => {
+	const [fileList, setFileList] = useState([]);
+	const [uploading, setUploading] = useState(false);
+
+	const handleUpload = () => {
+		if (fileList.length === 0) {
+			message.warning("Please select a file before uploading.");
+			return;
+		}
+
+		const formData = new FormData();
+		fileList.forEach((file) => {
+			formData.append("file", file);
+		});
+
+		setUploading(true);
+
+		fetch("https://660d2bd96ddfa2943b33731c.mockapi.io/api/upload", {
+			method: "POST",
+			body: formData,
+		})
+			.then((res) => {
+				if (!res.ok) throw new Error("Upload failed. Please try again.");
+				return res.json();
+			})
+			.then(() => {
+				setFileList([]); // Clear the file list after successful upload
+				message.success("Upload successful.");
+			})
+			.catch((error) => {
+				message.error(error.message || "Upload failed.");
+			})
+			.finally(() => {
+				setUploading(false);
+			});
+	};
+
+	const props = {
+		onRemove: (file) => {
+			setFileList((prevList) =>
+				prevList.filter((item) => item.uid !== file.uid),
+			);
+		},
+		beforeUpload: (file) => {
+			if (file.type !== "application/zip" && !file.name.endsWith(".zip")) {
+				message.error("You can only upload ZIP files.");
+				return false;
+			}
+			setFileList((prevList) => [...prevList, file]);
+			return true; // Allow the file to be added to the list
+		},
+		fileList,
+		accept: ".zip",
+	};
+
+	return (
+		<div
+			style={{
+				display: "flex",
+				justifyContent: "center",
+				alignItems: "center",
+				height: "100vh",
+				padding: "16px",
+				backgroundColor: "#f4f6f8",
+			}}
+		>
+			<Card
+				style={{
+					width: "100%",
+					maxWidth: 500, // ✅ Responsive fix: works on mobile & desktop
+					textAlign: "center",
+					padding: "24px",
+					borderRadius: "10px",
+					boxShadow: "0px 4px 12px rgba(0, 0, 0, 0.1)",
+				}}
+			>
+				<CloudUploadOutlined style={{ fontSize: "50px", color: "#1890ff" }} />
+
+				{/* 🔽 Updated Title + Instructions 🔽 */}
```

Comment:
I am a fan of good comments and I 💙 the emojis in code :)

But actually this comment isn't needed here. 

A good comment is one that helps future developer who are reading your code (or even yourself in future when you've forgotten why you did something!)